### PR TITLE
Snippets

### PR DESCRIPTION
**Pull request by gabryelreyes**
Added code Snippets and Templates, as well as format-on-save setting.

### CODE REVIEW

File: .vscode/templates.code-snippets [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+		"description": "Simple Doxygen comment."
+	},
+	"Doxygen Block Comment/Description": {
+		"prefix": "doxy_block",
+		"scope": "c, cpp",
+		"body": [
+			"/**",
+			" * ${1:description}",
+			" * ",
+			" * ${2:parameters}",
```
Comment (BlueAndi) [inline .vscode/templates.code-snippets:50]: usually one empty line between parameters and return description

File: .vscode/templates.code-snippets [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+		],
+		"description": "Doxygen comment/description block."
+	},
+	"Template Header File": {
+		"prefix": "template_header",
+		"scope": "c, cpp",
+		"body": [
+			"/* MIT License",
+			" *",
+			" * Copyright (c) 2023 Andreas Merkle <web@blue-andi.de>",
```
Comment (BlueAndi) [inline .vscode/templates.code-snippets:63]: Can the year be automatically set to the current year.

**Review by BlueAndi — CHANGES_REQUESTED** [review]

File: .vscode/templates.code-snippets (hunk 2 above)
Comment (gabryelreyes) [reply .vscode/templates.code-snippets:63]: Not automatically, no. We could make it variable, and each time we have to type the year. Or we change the template each year. Or make an script of some kind to run on the CI

Comment (gabryelreyes) [reply .vscode/templates.code-snippets:64]: Forget it, just found out how

**Review by BlueAndi — APPROVED** [review]

**Pull request merged**